napari/layers/utils/_color_manager_constants.py

```diff
@@ -0,0 +1,19 @@
+from enum import auto
+
+from ...utils.misc import StringEnum
+
+
+class ColorMode(StringEnum):
```

Comment:
this will now make the fourth time we've defined this exact same class:
1. [points](https://github.com/napari/napari/blob/0a75b4d840c06763cb6654d055cd450fd869cde8/napari/layers/points/_points_constants.py#L6)
2. [shapes](https://github.com/napari/napari/blob/0a75b4d840c06763cb6654d055cd450fd869cde8/napari/layers/shapes/_shapes_constants.py#L38)
3. [vectors](https://github.com/napari/napari/blob/0a75b4d840c06763cb6654d055cd450fd869cde8/napari/layers/vectors/_vectors_constants.py#L8)

is there no way to avoid this?  (cc #1492)

Reply:
@tlambert03 , sorry I wasn't clear in the PR description. The goal of this PR is to bring all of that duplicated code (including the multiply-defined `ColorMode`) into one place. Currently, I am still at the stage of exploring if the dataclass is the right way to create the `ColorManager` (as opposed to a standard class). Once that is working, I will use the `ColorManager` to set colors in Points, Shapes, and Vectors. Thus, `ColorMode` will be defined only here.

Reply:
Oh great!  Sorry, my bad :)

Reply:
No worries!